Проект ShareIt. Техническое задание. Часть 2

Akmula opened the pull request.

Akmula committed:
Проект ShareIt. Техническое задание. Часть 2

Akmula committed:
Проект ShareIt. Техническое задание. Часть 2

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

# Conflicts:
#	src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java
#	src/main/java/ru/practicum/shareit/item/ItemServiceImpl.java
#	src/main/java/ru/practicum/shareit/user/UserServiceImpl.java

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

# Conflicts:
#	src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java
#	src/main/java/ru/practicum/shareit/item/ItemServiceImpl.java
#	src/main/java/ru/practicum/shareit/user/UserServiceImpl.java

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

Akmula committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

VadimZharkov commented on src/main/java/ru/practicum/shareit/booking/BookingController.java:
```diff
@@ -1,9 +1,75 @@
+        } else {
+            log.info("Отклонение бронирования предмета владельцем: {}", userId);
+        }
+        BookingDtoResponse bookingDtoResponse = bookingService.approvedBooking(userId, bookingId, approved);
```
Достаточно назвать BookingResponse, из логики понятно что это Dto и слишком перегружать название нет смысла

Akmula replied:
Исправил.

VadimZharkov commented on src/main/java/ru/practicum/shareit/booking/BookingController.java:
```diff
@@ -1,9 +1,75 @@
+    @PatchMapping("/{bookingId}")
+    public BookingDtoResponse approvedBooking(@RequestHeader("X-Sharer-User-Id") int userId,
+                                              @PathVariable int bookingId,
+                                              @RequestParam Boolean approved) {
```
Можно использовать примитивный тип boolean

Akmula replied:
Исправил.

VadimZharkov commented on src/main/java/ru/practicum/shareit/booking/BookingRepository.java:
```diff
@@ -0,0 +1,57 @@
+    List<Booking> findAllBookingsByBookerIdOrderByStartDesc(int bookerId);
+
+    //Параметр state равен CURRENT.
+    @Query("""
```
Отлично что используешь текстовый блоки """ ... """

Akmula replied:
Спс. Мне так больше нравится)

VadimZharkov commented on src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java:
```diff
@@ -0,0 +1,201 @@
+        getUser(bookerId);
+        List<Booking> bookingDtoResponses;
+
+        switch (state.toUpperCase()) {
```
Для state лучше сделать перечислимое

Akmula replied:
Сделал enum State

VadimZharkov commented on src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java:
```diff
@@ -0,0 +1,201 @@
+                throw new UnknownStateException("Unknown state: " + state);
+            }
+        }
+        return bookingDtoResponses.stream().map(BookingMapper::toBookingDto).collect(Collectors.toList());
```
Для операторов стримов принято их переносить на новую строку для лучшей читаемости
```
bookingDtoResponses.stream()
    .map(BookingMapper::toBookingDto)
    .collect(Collectors.toList());
```

Akmula replied:
Исправил.

VadimZharkov commented on src/main/java/ru/practicum/shareit/item/ItemMapper.java:
```diff
@@ -0,0 +1,67 @@
+@Component
+public class ItemMapper {
+
+    public static Item dtoToItem(ItemDtoRequest itemDtoRequest, User user, Integer itemId) {
```
Eсли используешь `@Component`, то static лишнее потому что спринг создаст один объект ItemMapper на приложение и заинжектит его где надо.
 static имеет смысл, если бы ты ItemMapper  использовал бы без `@Component`

Akmula replied:
Исправил.

VadimZharkov commented on src/main/java/ru/practicum/shareit/item/ItemMapper.java:
```diff
@@ -0,0 +1,67 @@
+    public static Item dtoToItem(ItemDtoRequest itemDtoRequest, User user, Integer itemId) {
+        Item item = new Item();
+        if (itemId != null) {
+            item.setId(itemId);
```
Можно просто item.setId(itemId); без проверки на null,  т.к. здесь Item item = new Item();  id уже null
и тогда можно использовать `@Builder`

Akmula replied:
Исправил.

VadimZharkov commented on src/main/resources/schema.sql:
Здесь все красиво

VadimZharkov reviewed (changes requested):
Хорошая работа, можно улучшить

Akmula committed:
1. Убрать DTO в ответах на запросы
2. Вместо Boolean использовать boolean
3. Сделать enum State
4. Подправить переносы стримов
5. Убрать static в мапперах
